API and Gateway v8 (#2097)
hammerandchisel/discord-api-docs: docs/Change_Log.md

# Change Log

## API and Gateway V8

#### September 24, 2020

We've introduced API and Gateway v8! Changes are noted throughout the documentation, but the major changes are:

> - API v8 is now default

Reviewer: v8 is not yet default. we'll need to have a deprecation period before it hits default

- All permissions have been converted to strings
- `permissions_new`, `allow_new`, and `deny_new` have been removed
- The `game` field has been removed. You should instead reference the first element of the activites array
- Channel Permission Overwrites are now numbers (0 and 1) instead of strings ("role" and "member")
- `embed_enabled` and `embed_channel_id` have been renamed `widget_enabled` and `widget_channel_id`
- Form body errors have been improved to include more helpful messaging on validation. [See more here](#DOCS_TOPICS_OPCODES_AND_STATUS_CODES/API)
- Rate limit headers are in seconds instead of milliseconds. As such, the `X-RateLimit-Precision header` is no longer respected. If you receive any millisecond headers, please report it as a bug!
- Bots no longer receive [Channel Create Gateway Event](#DOCS_GATEWAY/channel-create) for DMs

> - `delete-message-days` is no longer available. Use `delete_message_days`. The previous value may never have been documented

Reviewer 2: was documented and updated in b9edace323c9df64c79f104d85984690ae4e2977

# Change Log

## API and Gateway V8

#### September 24, 2020

We've introduced API and Gateway v8! Changes are noted throughout the documentation, but the major changes are:

- API v8 is now available. v6 is still the default for the time being.
- All permissions have been converted to strings-serialized numbers. As such, `permissions_new`, `allow_new`, and `deny_new` have been removed
- The `game` field has been removed. If you need a direct replacement, you can instead reference the first element of `activities`
- Channel Permission Overwrite `type`s are now numbers (0 and 1) instead of strings ("role" and "member")

> - `embed_enabled` and `embed_channel_id` have been renamed `widget_enabled` and `widget_channel_id`

Reviewer 2: not really renamed, the `widget_` fields already existed
```suggestion
- `embed_enabled` and `embed_channel_id` have been removed. Use `widget_enabled` and `widget_channel_id` instead.
```

- Form body errors have been improved to include more helpful messaging on validation. [See more here](#DOCS_REFERENCE/error-messages)
- The `Retry-After` header is now based in seconds instead of milliseconds (e.g. `123` means 123 seconds)
- The `X-RateLimit-Precision` header is no longer respected. `X-RateLimit-Reset` and `X-RateLimit-Reset-After` are always returned at millisecond precision (e.g. `123.456` instead of `124`)
- Bots no longer receive [Channel Create Gateway Event](#DOCS_GATEWAY/channel-create) for DMs
- `delete-message-days` is no longer available. Use `delete_message_days`.
- Removed `roles`, `premium_since`, and `nick` from [Presence Update Gateway Event](#DOCS_GATEWAY/presence-update)

> - The following routes have been deprecated and removed for better naming conventions:

Reviewer 2: were already deprecated
```suggestion
- The following deprecated routes have been removed for better naming conventions:
```

Reviewer 2: Also, `/invite/<invite_code>` was removed in favor of `/invites/<invite_code>`

Removed in favor of `/guilds/<guild_id>/widget`:

- `/guilds/<guild_id>/embed`

Removed in favor of `/guilds/<guild_id>/widget.json`:

- `/servers/<guild_id>/embed.json`
- `/servers/<guild_id>/widget.json`
- `/guilds/<guild_id>/embed.json`

> Removed in favor if `/guilds/<guild_id>/widget.png`:

Reviewer 3: Typo. Should be `of` instead of `if`.

# Change Log

## API and Gateway V8

#### September 24, 2020

We've introduced API and Gateway v8! Changes are noted throughout the documentation, but the major changes are:

> - API v8 is now available. v6 is still the default for the time being.

Reviewer 4: ```suggestion
- API and Gateway v8 is now available. v6 is still the default for the time being.
```

- All permissions have been converted to strings-serialized numbers. As such, `permissions_new`, `allow_new`, and `deny_new` have been removed
- The `game` field has been removed. If you need a direct replacement, you can instead reference the first element of `activities`

> - Channel Permission Overwrite `type`s are now numbers (0 and 1) instead of strings ("role" and "member"). However due to a current technical constraint, they are string-serialized numbers in audit logs.

Reviewer 4: ```suggestion
- Channel Permission Overwrite `type`s are now numbers (0 and 1) instead of strings ("role" and "member"). However due to a current technical constraint, they are string-serialized numbers in audit log `options`.
```

- `embed_enabled` and `embed_channel_id` have been removed. Use `widget_enabled` and `widget_channel_id` instead.
- Form body errors have been improved to include more helpful messaging on validation. [See more here](#DOCS_REFERENCE/error-messages)
- The `Retry-After` header is now based in seconds instead of milliseconds (e.g. `123` means 123 seconds)
- The `X-RateLimit-Precision` header is no longer respected. `X-RateLimit-Reset` and `X-RateLimit-Reset-After` are always returned at millisecond precision (e.g. `123.456` instead of `124`)

> - Bots no longer receive [Channel Create Gateway Event](#DOCS_GATEWAY/channel-create) for DMs

Reviewer 5: I'm curious, what's the reasoning behind this change? Was it just getting that expensive to send these to bots constantly?

Reviewer 2: iirc gateway v6 sent a channel_create every time a dm was sent, even if in an existing dm

Reviewer 5: oooo right, I completely forgot about that >w<
